### PR TITLE
Expose compose task runner options

### LINKED ISSUE #4257: Expose compose task runner options
As ctr app is now defined within a server configuration and not as a registered app we don't have a direct way from UI to ask its properties. This relates to #1515 where for ctr task it'd be nice to show possible options for it.

We should add a new rest endpoint which exposes options similarly what we can do with registered apps.

### PR DESCRIPTION
**Pull request by jvalkeal**
- Add boot metadata to ctr jar
- Add controller /tasks/ctr/options to expose
  those from a server to get used by UI.
- Fixes #4257

### CODE REVIEW

**Comment by jvalkeal** [comment]
Right when starting from empty maven repo we don't yet have ctr jar with metadata when server-core is tested.

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskCtrController.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+			TaskConfigurationProperties taskConfigurationProperties,
+			ComposedTaskRunnerConfigurationProperties composedTaskRunnerConfigurationProperties,
+			AppResourceCommon appResourceCommon) {
+		this.metadataResolver = metadataResolver;
+		this.taskConfigurationProperties = taskConfigurationProperties;
+		this.composedTaskRunnerConfigurationProperties = composedTaskRunnerConfigurationProperties;
+		this.appResourceCommon = appResourceCommon;
+	}
+
+	@RequestMapping(value = "/options", method = RequestMethod.GET)
```
Comment (cppwfs) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskCtrController.java:59]: Probably need to add something like `root.add(WebMvcLinkBuilder.linkTo(WebMvcLinkBuilder.methodOn(TaskCtrController.class).options()).withRel("tasks/ctr/options"));` to https://github.com/spring-cloud/spring-cloud-dataflow/blob/master/spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/RootController.java#L176

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DataFlowControllerAutoConfiguration.java [hunk 2]
```diff
@@ -243,6 +245,15 @@ public AppRegistryController appRegistryController(
 					metadataResolver, appRegistryFJPFB, streamDefinitionService, appRegistrationAssemblerProvider);
 		}
 
+		@Bean
+		public TaskCtrController tasksCtrController(ApplicationConfigurationMetadataResolver metadataResolver,
```
Comment (cppwfs) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DataFlowControllerAutoConfiguration.java:249]: Would this fit better in the TaskEnabledConfiguration of the DataFlowControllerAutoConfiguration?   In that the CTR will not be active if tasks are not active.   Thoughts?

File: spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/TaskCtrControllerTests.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import static org.mockito.ArgumentMatchers.any;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = TestDependencies.class)
+@AutoConfigureTestDatabase(replace = Replace.ANY)
+public class TaskCtrControllerTests {
```
Comment (cppwfs) [inline spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/TaskCtrControllerTests.java:50]: @author tag.

File: spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/TaskCtrControllerTests.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+		p.setId("oauth2-client-credentials-scopes");
+		p.setName("oauth2-client-credentials-scopes");
+		List<ConfigurationMetadataProperty> props = Arrays.asList(p);
+		Mockito.when(metadataResolver.listProperties(any())).thenReturn(props);
+	}
+
+	@Test
+	public void testOptions() throws Exception {
+		mockMvc.perform(get("/tasks/ctr/options").accept(MediaType.APPLICATION_JSON))
+			.andDo(print())
```
Comment (cppwfs) [inline spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/TaskCtrControllerTests.java:74]: Probably can remove the print()

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskCtrController.java (hunk 1 above)
Comment (jvalkeal) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskCtrController.java:59]: I think we want to keep this away from a "public eye" as its solely made for UI.

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DataFlowControllerAutoConfiguration.java (hunk 2 above)
Comment (jvalkeal) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DataFlowControllerAutoConfiguration.java:249]: Yeah, you're right.

**Comment by cppwfs** [comment]
LGTM

**Comment by cppwfs** [comment]
Rebased, squashed, Merged.

**Pull request closed without merging**